modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -872,33 +911,92 @@ func (p *ProjectPipelineService) batchGetPipelineSources(pipelineSourceIDArray [
 	return pipelineSourceMap, nil
 }
 
-func (p *ProjectPipelineService) autoRunPipeline(identityInfo apistructs.IdentityInfo, definition *dpb.PipelineDefinition, source *spb.PipelineSource) (*apistructs.PipelineDTO, error) {
-	var extraValue = apistructs.PipelineDefinitionExtraValue{}
-	err := json.Unmarshal([]byte(definition.Extra.Extra), &extraValue)
+func (p *ProjectPipelineService) autoRunPipeline(identityInfo apistructs.IdentityInfo, definition *dpb.PipelineDefinition, source *spb.PipelineSource, appDto *apistructs.ApplicationDTO, rules []*apistructs.BranchRule) (*apistructs.PipelineDTO, error) {
+	extraValue, err := getExtraValue(definition)
 	if err != nil {
 		return nil, apierrors.ErrRunProjectPipeline.InternalError(fmt.Errorf("failed unmarshal pipeline extra error %v", err))
 	}
+	createV2 := extraValue.CreateRequest
 
 	if err := p.checkRolePermission(identityInfo, extraValue.CreateRequest, apierrors.ErrRunProjectPipeline); err != nil {
 		return nil, err
 	}
 
-	createV2 := extraValue.CreateRequest
+	validBranch := diceworkspace.GetValidBranchByGitReference(createV2.Labels[apistructs.LabelBranch], rules)
+	workspace := validBranch.Workspace
+	createV2.Labels[apistructs.LabelDiceWorkspace] = workspace
+
+	// update user gittar token
+	var worker = limit_sync_group.NewWorker(3)
+	worker.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
+		orgStr := createV2.Labels[apistructs.LabelOrgID]
+		orgID, err := strconv.ParseUint(orgStr, 10, 64)
+		if err != nil {
+			return apierrors.ErrRunProjectPipeline.InternalError(fmt.Errorf("not find orgID"))
+		}
+		// update CmsNsConfigs
+		if err = p.UpdateCmsNsConfigs(identityInfo.UserID, orgID); err != nil {
+			return apierrors.ErrRunProjectPipeline.InternalError(err)
+		}
+		return nil
+	})
+	// update createV2 LabelCommitDetail info
+	worker.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
+		// get newest commit info
+		commit, err := p.bundle.GetGittarCommit(appDto.GitRepoAbbrev, createV2.Labels[apistructs.LabelBranch], identityInfo.UserID)
+		if err != nil {
+			return apierrors.ErrGetGittarCommit.InternalError(err)
+		}
+		detail := apistructs.CommitDetail{
+			CommitID: commit.ID,
+			Repo:     appDto.GitRepo,
+			RepoAbbr: appDto.GitRepoAbbrev,
+			Author:   commit.Committer.Name,
+			Email:    commit.Committer.Email,
+			Time:     &commit.Committer.When,
+			Comment:  commit.CommitMessage,
+		}
+		commitInfo, err := json.Marshal(&detail)
+		if err != nil {
+			return err
+		}
+
+		locker.Lock()
+		defer locker.Unlock()
+
+		createV2.Labels[apistructs.LabelCommitDetail] = string(commitInfo)
+		return nil
+	})
+	// update createV2 ConfigManageNamespaces
+	worker.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
+		relationMap, err := p.AppSvc.GetPublishItemRelationsMap(apistructs.QueryAppPublishItemRelationRequest{AppID: int64(appDto.ID)})
```

Comment:
strange, why need update publish-item's cms namespaces?

Reply:
Changed back to the original ConvertPipelineToV2 method for reuse